sumi/config.txt
# Client configuration file
# Saved by $Id$
# Please note - ALL COMMENTS IN THIS FILE WILL BE DESTROYED
{'allow_local': True,
 'allow_multicast': True,
 'bandwidth': 56623104,
 'bandwidth1': 1500000,
 'colwidths': [160, 107, 56, 55, 146, 69, 76, 99, 71, 71, 71, 39],
 'control_protocol': 'ack',
 'crypt': False,
 'data_chan_type': 'u',
 'dchanmode': 'socket',
 'dl_dir': '/home/jeff/sumi/trunk/sumi/incoming',
 'interface': '\\Device\\NPF_{5A98A9B4-4DCD-4E1B-A2B2-70913B25E61B}',
 'interface1': '\\Device\\NPF_{FDA97C53-F88D-486E-82B6-FBBD969F0F59}',
 'interface2': 'ndis0',
 'interface3': '\\Device\\NPF_{3530F223-8959-4928-88A5-1672C094D6A8}',
 'irc_nick': 'sumigex',
 'maxwait': 7,
 'mcast_iface': '0.0.0.0',
 'mtu': 1500,
 'multicast_groups': [],
 'multicast_groups2': [('224.51.105.104', None)],
 'myip': '10.0.0.106',
 'myip3': '10.0.0.64',
 'myport': 3490,
 'rcsid': '$Id',
 'redundancy': 20.0,
 'rwinsz': 5,
 'save_interval': 1,
 'share': False,
 'winsize': (1062, 190)}


sumi/ChangeLog
SUMI ChangeLog 
$Id$
vi: noexpandtab:

2006-10-18 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* r309 Wrap zlib.crc32() in my_crc32(), consistent across 32- and 64-bit

2006-10-16 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* r305 Configuration changes to use server on amd64 laptop

2006-10-04 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Research project started, will continue for Fall 2006 quarter

2005-09-16 Jeff Connelly <jeffconnelly@users.sourceforge.net> - 0.8.15
	* r290 Move to Python 2.4, see [[Python 2.4 migration]] for details
	* r289 Fix transports that rely on capture()
	* r288 Improve interface selection
	* r285 library.zip is now compressed
	* Win32 no-installer distribution is a smaller self-extracting 7z archive
	  (created manually)
	* Development of SUMI will slow down due to school, but I still plan to
	  work on it when I have time. Anyone want to help?

2005-09-15 Jeff Connelly <jeffconnelly@users.sourcefore.net>
	* Add "-e filename" flag to sumigetw for running arbitrary Python scripts
	* Turn on SO_REUSEADDR and (if available) SO_REUSEPORT
	* Client can join multicast groups
	* Server can choose which interface to multicast on

2005-09-12 Jeff Connelly <jeffconnelly@users.sourceforge.net> - 0.8.14
	* r257 Installer shows license, uninstaller works better,
	  it can uninstall WinPcap and mIRC transport script
	* The source code release is much smaller since it mostly only
	  contains what is in Subversion - none of the extra cruft
	* Open config if no args, remove allow_no_req

2005-09-11 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* r242 Whole-file SHA-1 hash for verifying integrity

2005-09-10 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* r234 CRC32 check in every packet
	* r233 32-bit seqno, allowing files up to ~5TB depending on MTU
	* Note: r### in this ChangeLog will refer to Subversion revision

2005-09-07 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Add another Win32 distribution without an installer (win32noinst
	  target), released as 0.8.13b with all changes below.
	* Installer runs uninstaller if an old version already exists (#1284284)
	* Don't include old HTML documentation with distribution; latest docs
	  are on wiki and should eventually be converted and distributed with SUMI

2005-09-06 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Added doc/wdoc.py to extract articles from SUMIWiki, pipe to wt2db
	* In server, if 'interface' is blank, autoselect using use_new_if(),
	  correcting "Pcap: Error opening : pcapy.PcapError" (thanks peersurfer)

2005-09-05 Jeff Connelly <jeffconnelly@users.sourceforge.net> - 0.8.13a
	* Added a few doctests
	* SUMI is listed on Slyck.com!
	* This is a source-only release; there are only minor changes from
	  the 0.8.13 binary Win32 release. 

2005-09-01 Jeff Connelly <jeffconnelly@users.sourcforge.net> - 0.8.13

2005-08-31 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Drag-and-drop support in GUI (try dropping a .sumi file into it)
	* Allow modirclib to be used with crypt_req
	* Rename config.py to config.txt

2005-08-29 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Add support for passing a .sumi filename to sumigetw
	* New .sumi file format
	* GUI "Resume" works on multiple transfers at once
	* Installer compressed with LZMA solid
		Default compresion: 4493015, vs. LZMA solid: 3330921
	  Without WinPcap, it is the smallest sumiinst of the 0.8 series
	* WinPcap 3.1 is now bundled with the installer
	* Installer sets up file associations

2005-08-25 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Fix abort, add GUI option to resume, and 'startfile' option
	
2005-08-22 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* sumigetw uses wxPython 2.6.1.0 and new-style symbol names, no more 2.4
	* Client configures MTU instead of MSS, make MSS not include header
	* Fix sumigetw crash after 30,000 characters were logged, add log files

2005-08-21 Jeff Connelly <jeffconnelly@users.sourceforge.net> - 0.8.12
	* New transport: modtor, uses The Onion Router (see http://tor.eff.org)
	* File data encryption fully works, enabled by default (crypt_req not)
	* Many minor bugfixes, removed modpipe, use shorter variable names ('u')

2005-08-11 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Partially autoconfigure IP, MSS, src_allow, and interface if
	  no 'interface' in config.py or no 'src_allow' in sumiserv.cfg.

2005-08-10 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* General code cleanup (pychecker again)
	* NAK lost packets are compressed, ex: 1-5,8,10 instead of 1,2,3,..

2005-07-31 Jeff Connelly <jeffconnelly@users.sourceforge.net> - 0.8.11
	* Auth packet includes hash of nonce for client to verify
	* Completely redone crypto implementation from 0.8.10
	* Use SHA-1 for key derivation function as recommended
	* Changed MD5 to SHA-1 in auth packet (MD5 still used for rawproxd)
	* Added INTERLOCK_DELAY; a time-sensitive enhancement to the Interlock
	  Protocol in order to detect some MITM

2005-07-23 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Request messages can now be encrypted, set "crypto" to "y"
	* EC-DH + interlock protocol for key exchange/request
	* AES-256 for symmetric encryption 
	* Protects against passive eavesdropping and active MITM (not hijack)
	* Works well, but incomplete--need to verify nonce, encrypt data

2005-07-14 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Removed cryptography for now--needs to be redone.

2005-07-08 Jeff Connelly <jeffconnelly@users.sourceforge.net>
	* Use log() instead of print in client, server, and transports
	* sumiget logs to "Client" tab, now works
	* Removed "Networks" tab
	* Fixed most pychecker warnings on source code
	* Split larger functions into smaller functions
	* Added 'noise' option to allow simulated packet loss
	* sumigetw now runs sumiserv in "Server" tab

2005-07-02 Jeff Connelly <jeffconnelly@users.sourceforge.net> - 0.8.10

sumi/config.py
# Client configuration file
# Saved by $Id$
# Please note - ALL COMMENTS IN THIS FILE WILL BE DESTROYED
{'allow_local': True,
 'allow_no_req': True,
 'bandwidth': 56623104,
 'bandwidth1': 1500000,
 'colwidths': [160, 130, 33, 55, 115, 69, 76, 99, 71, 49, 75, 39],
 'crypt': True,
 'data_chan_type': 'u',
 'dchanmode': 'socket',
 'dl_dir': '/mnt/smb/p2p/sumi/incoming',
 'interface': '\\Device\\NPF_{FDA97C53-F88D-486E-82B6-FBBD969F0F59}',
 'interface2': 'ndis0',
 'interface3': '\\Device\\NPF_{3530F223-8959-4928-88A5-1672C094D6A8}',
 'irc_chans': {'#sumi': 'anon'},
 'irc_nick': 'sumiget',
 'irc_port': 6667,
 'irc_server': 'irc.xtc-dreams.com',
 'maxwait': 7,
 'mtu': 1500,
 'myip': '',
 'myip2': '10.0.0.64',
 'myip3': '10.0.0.64',
 'myport': 3490,
 'passwd': '11111111111111111111111111111111',
 'rcsid': '$Id',
 'rwinsz': 5,
 'save_interval': 1,
 'share': False,
 'winsize': (977, 240)}
